templates/search_results.html

```diff
@@ -35,16 +36,34 @@ <h6 class="card-subtitle mb-2 text-muted">Number of ingredients:
                         </a>
                     </li>
                     {% endif %}
-                    {% for i in range(0, total_pages+1)%}
-                    <li class="page-item"><a class="page-link" href="{{i}}">{{i+1}}</a></li>
-                    {% endfor %}
-                    {% if cur_page != total_pages %}
-                    <li class="page-item">
-                        <a class="page-link" href="{{cur_page+1}}" aria-label="Next">
-                            <span aria-hidden="true">&raquo;</span>
-                        </a>
-                    </li>
-                    {% endif %}
+                    {% if total_pages <= 7 %} {% for i in range(0, total_pages+1)%} <li
```

Comment:
"Magic number" 7 - would be good to refactor into a constant.

Reply:
Also, the structure from here down is begging to be refactored into smaller templates.

Reply:
That's the pagination logic.

7 *is* a magic number, because that's what fits best on the page. Using flask variables is even less pretty, and almost harder to understand, but if you think I should in this case I will.

As to the rest of the code below it, that's all the pagination logic. It's to to keep the current page highlight closer to the beginning when you start going through pages, keep it centered and let the pages move around it when you're going through the middle, and then have the highlight move to the end as you go through the last few pages.

It's an ugly bit of code that makes the page look prettier.